scripts/project-catalogue/populate.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import json
+import os
+from typing import List, Tuple
+
+from requests.models import HTTPError
+
+from excel.load import SheetLoader, Entity
+from convert.tracker import DatasetTrackerConverter
+from convert.europe_pmc import EuropePmcConverter
+from services.europe_pmc import EuropePmc
+from services.ingest import QuickIngest
+
+SHEET = "Group 1"
+URL = "https://api.ingest.dev.archive.data.humancellatlas.org"
+TOKEN = "source/token.txt"
+
+
+class Populate:
+    def __init__(self, file_path, sheet=SHEET, url=URL, token_path=TOKEN):
+        self.ingest = QuickIngest(url, token=self.get_token(token_path))
+        self.path = file_path
+        self.sheet = sheet
+        loader = SheetLoader(self.path, self.sheet)
+        self.submission = loader.data
+        self.europe_pmc = EuropePmc()
+        self.sheet_converter = DatasetTrackerConverter()
+        self.publication_converter = EuropePmcConverter()
+
+    def sort_projects(self) -> Tuple[List[Entity], List[Entity]]:
+        new_projects = []
+        existing_projects = []
+        for project in self.submission.get_entities(self.sheet):
+            doi = project.attributes.get('doi', '')
+            project_uuid = project.attributes.get('ingest_project_uuid', '')
+            if (
+                    (doi and self.ingest.doi_exists(doi)) or
+                    (project_uuid and self.ingest.project_exists(project_uuid))
+            ):
+                existing_projects.append(project)
+            else:
+                new_projects.append(project)
+        return new_projects, existing_projects
+
+    def post_projects(self, projects: List[Entity]):
+        for project in projects:
+            conversion = self.convert_project(project)
+            self.post_project(project, conversion)
+
+    def convert_project(self, project: Entity):
+        conversion = self.sheet_converter.convert(project.attributes)
+        self.add_publication_info(conversion, project)
+        self.set_defaults(conversion, project)
+        return conversion
+
+    def add_publication_info(self, conversion: dict, project: Entity):
+        doi = project.attributes.get('doi', '')
+        if doi:
+            publication_info = self.europe_pmc.query_doi(doi)
+            if publication_info:
+                publications, info = self.publication_converter.convert(publication_info)
+                conversion.setdefault('content', {}).update(publications)
+                conversion['publicationsInfo'] = info
+
+    @staticmethod
+    def set_defaults(conversion: dict, project: Entity):
+        default_title = project.attributes.get('pub_title', '')
+        conversion.setdefault('content', {}).setdefault('project_core', {}).setdefault('project_title', default_title)
+        conversion.setdefault('content', {}).setdefault('project_core', {}).setdefault('project_short_name', project.identifier.index)
```

Comment:
Could a simpler way be:

```suggestion
        conversion['content']['project_core'].setdefault('project_short_name', project.identifier.index)
```

Reply:
nope, that would overwrite the content or project core dictionaries rather than adding to them